apps/api/src/app/change/usecases/promote-layout-change/promote-layout-change.ts

```diff
@@ -0,0 +1,42 @@
+import { Injectable } from '@nestjs/common';
+import { LayoutEntity, LayoutRepository } from '@novu/dal';
+
+import { PromoteTypeChangeCommand } from '../promote-type-change.command';
+
+@Injectable()
+export class PromoteLayoutChange {
+  constructor(private layoutRepository: LayoutRepository) {}
+
+  async execute(command: PromoteTypeChangeCommand) {
+    const item = await this.layoutRepository.findOne({
+      _environmentId: LayoutRepository.convertStringToObjectId(command.environmentId),
+      _id: command.item._id,
```

Comment:
To me it seems that it should be searched with _parentId instead of _id. So it checks if a layout that has already been promoted exists. 
_parentId is the id of the item in a previous environment (dev)